pulp_2to3_migration/tests/functional/test_rpm_repo.py

```diff
@@ -0,0 +1,212 @@
+import os
+import time
+import unittest
+
+from pulpcore.client.pulpcore import Configuration
+from pulpcore.client.pulp_rpm import (
+    ApiClient as RpmApiClient,
+    ContentAdvisoriesApi,
+    # ContentDistributionTreesApi,
+    ContentModulemdDefaultsApi,
+    ContentModulemdsApi,
+    ContentPackagecategoriesApi,
+    ContentPackageenvironmentsApi,
+    ContentPackagegroupsApi,
+    ContentPackagelangpacksApi,
+    ContentPackagesApi,
+    DistributionsRpmApi,
+    PublicationsRpmApi,
+    RemotesRpmApi,
+    RepositoriesRpmApi,
+    RepositoriesRpmVersionsApi,
+)
+from pulpcore.client.pulp_2to3_migration import (
+    ApiClient as MigrationApiClient,
+    MigrationPlansApi,
+)
+from pulp_2to3_migration.tests.functional.util import get_psql_smash_cmd, set_pulp2_snapshot
+
+from pulp_smash import cli
+from pulp_smash import config as smash_config
+from pulp_smash.pulp3.bindings import monitor_task, monitor_task_group
+
+from .common_plans import RPM_SIMPLE_PLAN, RPM_COMPLEX_PLAN
+from .constants import BINDINGS_CONFIGURATION, FIXTURES_BASE_URL, TRUNCATE_TABLES_QUERY_BASH
+
+PULP_2_RPM_DATA = {
+    'repositories': 5,
+    'remotes': 3,
+    'publications': 5,
+    'distributions': 5,
+    'content': {
+        'rpm-empty': {},
+        'rpm-empty-for-copy': {},
+        'rpm-with-modules': {
+            'advisory': 6,
+            'modulemd': 10,
+            'modulemd-defaults': 3,
+            'category': 1,
+            'group': 2,
+            'langpack': 1,
+            'package': 35,
+        },
+        'rpm-distribution-tree': {
+            'disttree': 1,
+            'environment': 1,
+            'category': 1,
+            'group': 1,
+            'langpack': 1,
+            'package': 1,
+        },
+        'srpm-unsigned': {
+            'advisory': 2,
+            'category': 1,
+            'group': 2,
+            'langpack': 1,
+            'package': 3,
+        },
+    },
+    'content_total': {
+        'package': 38,
+        'advisory': 8,
+        'modulemd': 10,
+        'modulemd-defaults': 3,
+        'disttree': 1,
+        'environment': 1,
+        'category': 3,
+        'group': 5,
+        'langpack': 3,
+    },
+
+}
+
+
+class BaseTestRpmRepo:
+    """
+    Test RPM repo, importer and distributor migration.
+    """
+    smash_cfg = smash_config.get_config()
+    smash_cli_client = cli.Client(smash_cfg)
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Create all the client instances needed to communicate with Pulp.
+        """
+        configuration = Configuration(**BINDINGS_CONFIGURATION)
+
+        rpm_client = RpmApiClient(configuration)
+        migration_client = MigrationApiClient(configuration)
+
+        # Create api clients for all resource types
+        cls.rpm_repo_api = RepositoriesRpmApi(rpm_client)
```

Comment:
We could probably make this class "generic" too, and re-use it for other plugins.

```python
 class RpmMigrationTest(MigrationPlanTestBase): 
    plugin_client = RpmApiClient(configuration)
    plugin_repo_api = RepositoriesRpmApi(rpm_client)
    plugin_repo_versions_api = RepositoriesRpmVersionsApi(rpm_client)
    ....
    PULP_2_DATA = {
        ....
    }

```